### PR TITLE
Registration bug

### PR DESCRIPTION
**Pull request by ppietraszko**
-Updated form and email validations so that if an email is already in use you can not register using the same one
-If a username is already in use you can not register using the same one
-As admissions staff you cannot register using a supervisor email that is the same as the email being used to register. 
- removed phone number from student and admin user models as it is not necessary and was causing issues with manually creating users

### CODE REVIEW

**Comment by ppietraszko** [comment]
I have updated the PR to include updated test cases

File: register/forms.py [hunk 1]
```diff
@@ -33,6 +33,12 @@ def __init__(self, *args, **kwargs):
             {"placeholder": "Password (Once more)"}
         )
 
+    def clean_username(self):
+        username = self.cleaned_data["username"]
+        if Student.objects.filter(username=username).exists():
```
Comment (snehalkenjale) [inline register/forms.py]: you can use get() as it should return only one object as username is unique.

File: register/validators.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+from django.core.exceptions import ValidationError
+
```
Comment (snehalkenjale) [inline register/validators.py]: import models here

Comment (ppietraszko) [reply register/validators.py]: it is imported with in the definition because otherwise there is an import error

File: register/forms.py (hunk 1 above)
Comment (ppietraszko) [reply register/forms.py]: when I switched it to get() a lot of the tests failed. any idea why?

Comment (snehalkenjale) [reply register/forms.py]: Get is meant to return single objects mostly used when to filter on primary key, where as filter() returns multiple objects. You will need to modify the test cases accordingly

File: register/validators.py (hunk 2 above)
Comment (snehalkenjale) [reply register/validators.py]: I meant instead of writing the import statement in the methods. Write it over here.

Comment (ppietraszko) [reply register/validators.py]: if you write it up top then you get an import error for some reason but importing in the method makes it work.

**Comment by abe-shek** [comment]
The changes include a ToDo comment for getting the review done, in the next sprint. Merging this since it's an critical feature for the demo.

**Pull request merged**